### PR TITLE
Add dynamic metric alert api

### PR DESCRIPTION
**Pull request by meday89**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5861 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5861/commits/2c7a862c8d33a544e9f2eabd874e9e539e960c6f

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json [hunk 1]
```diff
@@ -537,14 +541,14 @@
         "allOf": {
           "type": "array",
           "items": {
-            "$ref": "#/definitions/MetricCriteria"
+            "$ref": "#/definitions/StaticMetricCriteria"
```
Comment (shahabhijeet) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json]: Breaking change in stable API.
Any reason you would like to change your models in a stable version?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/examples/createOrUpdateDynamicMetricAlertSingleResource.json [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      "api-version": "2018-03-01",
+      "parameters": {
+        "location": "global",
+        "tags": {},
+        "properties": {
+          "description": "This is the description of the rule1",
+          "severity": 3,
+          "enabled": true,
+          "scopes": [
+            "/subscriptions/14ddf0c5-77c5-4b53-84f6-e1fa43ad68f7/resourceGroups/gigtest/providers/Microsoft.Compute/virtualMachines/gigwadme"
```
Comment (shahabhijeet) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/examples/createOrUpdateDynamicMetricAlertSingleResource.json]: @meday89 if you are using real subscriptionId, use a fake subscriptionId.
We recommend (000000-0000-00000...…..)

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json (hunk 1 above)
Comment (meday89) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/metricAlert_API.json]: Fixed, please see updates.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/examples/createOrUpdateDynamicMetricAlertSingleResource.json (hunk 2 above)
Comment (meday89) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-03-01/examples/createOrUpdateDynamicMetricAlertSingleResource.json]: Fixed, please see updates.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**